GLOSSARY.md

```diff
@@ -47,7 +47,8 @@ positive literal. A *Definite clause* has exactly one positive literal.
 *Horn clauses* can be written in the form `A || !B || !C || ...` with `A` being
 the optional positive literal. Due to the equivalence `(P => Q) <=> (!P || Q)`
 the clause can be expressed as `B && C && ... => A` which means that A is true
-if `B`, `C`, etc. are all true. All rules in chalk are in this form. For example
+if `B`, `C`, etc. are all true. All rules in *chalk* are in this form. For
```

Comment:
This is not, strictly speaking, true. The rules in Chalk are actually "hereditary harrop" clauses, which is an extension of Horn clauses. For example, Horn clauses do not support `forall<T> { ... }` goals or `A => B` goals.

Reply:
I added a short paragraph about hereditary Harrop clauses. But I wonder about the second type of goal that you mention. It is not yet clear to me why it is not supported by Horn clauses.

Reply:
>  It is not yet clear to me why it is not supported by Horn clauses.

A `A => B` goal means "assume `A` is true, then prove `B`". Horn clauses simply have no such construct. You have to create *global* rules saying what is and is not provable, and then apply them everywhere (vs having a local hypothesis that `A` is true which is only usable while proving `B`).

Reply:
(Having some trouble with the GitHub UI... accidentally deleted my comment from 15 minutes ago.) I think, I have still a bit trouble understanding. I'm quite sure that  `(P => Q) <=> (!P || Q)`. Therefore the "effect" with respect to the unification for both kinds of clauses has to be  different, right? Do you mean that for *Horn* clauses you have to create *global* rules, whereas for *HH* clauses you can formulate *local* rules? What does that mean exactly? Could you please elaborate a bit more? Thanks :)